libc-bottom-half/sources/chdir.c

```diff
@@ -0,0 +1,70 @@
+#include <errno.h>
+#include <fcntl.h>
+#include <limits.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <wasi/libc-find-relpath.h>
+#include <wasi/libc.h>
+
+char *__wasilibc_cwd = "/";
+static int __wasilibc_cwd_mallocd = 0;
```

Comment:
Since PWD is defined to be shared between threads better to leave it a non-TLS for now and include that error message.

Once we support threads we can most likely use some kind of atomic or lock here.

Reply:
Oops, yeah, `__wasilibc_cwd` is shared; I meant to add this comment to the `relative_buf` variables.

Reply:
I've gone ahead and added a compile time error because this'll definitely all need quite a lot of synchronization.